src/main/java/io/reactivex/Flowable.java

```diff
@@ -8877,14 +8877,14 @@ public final Disposable forEachWhile(final Predicate<? super T> onNext, final Co
      *  <dd>{@code last} does not operate by default on a particular {@link Scheduler}.</dd>
      * </dl>
      *
-     * @return a Flowable that emits the last item from the source Publisher or notifies Subscribers of an
+     * @return a Single that emits the last item from the source Publisher or notifies Subscribers of an
      *         error
      * @see <a href="http://reactivex.io/documentation/operators/last.html">ReactiveX operators documentation: Last</a>
      */
     @BackpressureSupport(BackpressureKind.UNBOUNDED_IN)
     @SchedulerSupport(SchedulerSupport.NONE)
-    public final Flowable<T> last() {
-        return takeLast(1).single();
+    public final Single<T> last() {
```

Comment:
The javadoc clearly states that an empty stream results in `NoSuchElementException`; but the former use of `single()` here is also a telltale sign.